crates/mcp-client/src/transport/stdio.rs

```diff
@@ -103,18 +104,25 @@ impl StdioActor {
 pub struct StdioTransport {
     command: String,
     args: Vec<String>,
+    env: HashMap<String, String>,
 }
 
 impl StdioTransport {
-    pub fn new<S: Into<String>>(command: S, args: Vec<String>) -> Self {
+    pub fn new<S: Into<String>>(
+        command: S,
+        args: Vec<String>,
+        env: HashMap<String, String>,
+    ) -> Self {
         Self {
             command: command.into(),
             args,
+            env: env,
         }
     }
 
     async fn spawn_process(&self) -> Result<(Child, ChildStdin, ChildStdout), Error> {
         let mut process = Command::new(&self.command)
+            .envs(&self.env)
```

Comment:
Will this do any validation for us to make sure the values passed are valid as environment vars? Names etc. Would be nice to get some descriptive messages if anything goes wrong.

Reply:
this totally depends on the server implementation. in the case of the Tavily example: it errors if `TAVILY_API_KEY` is not [set](https://github.com/RamXX/mcp-tavily/blob/12dfe964c44fb55a7021181be8ec4985ba185486/src/mcp_server_tavily/__init__.py#L24) but it won't error if its an invalid key